pkg/binding/binding.go

```diff
@@ -0,0 +1,217 @@
+}
+
+func (o *BindingClient) AskBindingName(serviceName, componentName string, flags map[string]string) (string, error) {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	defaultName := fmt.Sprintf("%v-%v", componentName, strings.Split(serviceName, " ")[0])
+	return backend.AskBindingName(defaultName, flags)
+}
+
+func (o *BindingClient) AskBindAsFiles(flags map[string]string) (bool, error) {
+	var backend backend.CreateBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+	return backend.AskBindAsFiles(flags)
+}
+
+func (o *BindingClient) CreateBinding(serviceName string, bindingName string, bindAsFiles bool, obj parser.DevfileObj, serviceMap map[string]servicebinding.Ref, componentContext string) error {
+	// serviceName format is <name> (<kind>.<apigroup>)
+	serviceRef := serviceMap[serviceName]
+	gvr, err := o.kubernetesClient.GetRestMappingFromGVK(schema.GroupVersionKind{
+		Group:   serviceRef.Group,
+		Version: serviceRef.Version,
+		Kind:    serviceRef.Kind,
+	})
+	if err != nil {
+		return err
+	}
+
+	service := servicebinding.Service{
+		Id: &bindingName, // Id field is helpful if user wants to inject mappings (custom binding data)
+		NamespacedRef: servicebinding.NamespacedRef{
+			Ref: servicebinding.Ref{
+				Group:    gvr.GroupVersionKind.Group,
+				Version:  gvr.GroupVersionKind.Version,
+				Kind:     gvr.GroupVersionKind.Kind,
+				Name:     strings.Split(serviceName, " ")[0],
+				Resource: gvr.Resource.Resource,
+			},
+		},
+	}
+
+	deploymentGVR, err := o.kubernetesClient.GetDeploymentAPIVersion()
+	if err != nil {
+		return err
+	}
+
+	serviceBinding := &servicebinding.ServiceBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: strings.Join([]string{kclient.ServiceBindingGroup, kclient.ServiceBindingVersion}, "/"),
+			Kind:       kclient.ServiceBindingKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: bindingName,
+		},
+		Spec: servicebinding.ServiceBindingSpec{
+			DetectBindingResources: true,
+			BindAsFiles:            bindAsFiles,
+			Application: servicebinding.Application{
+				Ref: servicebinding.Ref{
+					Name:     fmt.Sprintf("%s-app", obj.GetMetadataName()),
+					Group:    deploymentGVR.Group,
+					Version:  deploymentGVR.Version,
+					Resource: deploymentGVR.Resource,
+				},
+			},
+			Mappings: []servicebinding.Mapping{},
+			Services: []servicebinding.Service{service},
+		},
+	}
+
+	serviceBindingUnstructured, err := kclient.ConvertK8sResourceToUnstructured(serviceBinding)
+	if err != nil {
+		return err
+	}
+
+	yamlDesc, err := yaml.Marshal(serviceBindingUnstructured.UnstructuredContent())
+	if err != nil {
+		return err
+	}
+
+	err = libdevfile.AddKubernetesComponentToDevfile(string(yamlDesc), serviceBinding.Name, obj)
+	return err
+}
+
+func (o *BindingClient) GetServiceInstances() ([]string, map[string]servicebinding.Ref, error) {
```

Comment:
It seems this function could be part of a `kclient/binding.go` file

Reply:
https://github.com/redhat-developer/odo/pull/5641#discussion_r867064683